### PR TITLE
.github: Enable windows binary builds (excluding conda)

### PR DESCRIPTION
**Pull request by seemethere**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #71487
* __->__ #71484

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

This excludes conda, see: https://github.com/pytorch/pytorch/pull/71484#issuecomment-1022617195

Differential Revision: [D33800904](https://our.internmc.facebook.com/intern/diff/D33800904)

### CODE REVIEW

**Comment by seemethere** [comment]
Going to not include conda for this first pass through since it's currently failing with the following error:

```
/c/actions-runner/_work/pytorch/pytorch/builder/tmp_conda_3.7_234154/conda/Scripts/conda.exe install -y out_py3.7_cpu_0_20220125/win-64/pytorch-1.11.0.dev20220125-py3.7_cpu_0.tar.bz2

CondaVerificationError: The package for pytorch located at C:\actions-runner\_work\pytorch\pytorch\builder\tmp_conda_3.7_234154\conda\pkgs\pytorch-1.11.0.dev20220125-py3.7_cpu_0
appears to be corrupted. The path 'Lib/site-packages/caffe2/python/serialized_test/data/operator_test/collect_and_distribute_fpn_rpn_proposals_op_test.test_collect_and_dist.zip'
specified in the package manifest cannot be found.

CondaVerificationError: The package for pytorch located at C:\actions-runner\_work\pytorch\pytorch\builder\tmp_conda_3.7_234154\conda\pkgs\pytorch-1.11.0.dev20220125-py3.7_cpu_0
appears to be corrupted. The path 'Lib/site-packages/caffe2/python/serialized_test/data/operator_test/learning_rate_adaption_op_test.test_learning_rate_adaption_op_normalization.zip'
specified in the package manifest cannot be found.

CondaVerificationError: The package for pytorch located at C:\actions-runner\_work\pytorch\pytorch\builder\tmp_conda_3.7_234154\conda\pkgs\pytorch-1.11.0.dev20220125-py3.7_cpu_0
appears to be corrupted. The path 'Lib/site-packages/caffe2/python/serialized_test/data/operator_test/piecewise_linear_transform_test.test_multi_predictions_params_from_arg.zip'
specified in the package manifest cannot be found.
```

Interestingly enough this error doesn't appear to manifest itself when attempting to install the binary on a separate windows machine. CUDA binaries also exhibit similar behavior with an even longer list of missing zip files.

Link to example binary: https://gha-artifacts.s3.amazonaws.com/pytorch/pytorch/1748237956/conda-py3_7-cpu/pytorch-1.11.0.dev20220125-py3.7_cpu_0.tar.bz2

**Comment by seemethere** [comment]
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33800904).

**Review by malfet — APPROVED** [review]

File: .github/templates/windows_binary_build_workflow.yml.j2 [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+  ALPINE_IMAGE: "308535385114.dkr.ecr.us-east-1.amazonaws.com/tool/alpine"
+  ANACONDA_USER: pytorch
+  AWS_DEFAULT_REGION: us-east-1
+  BUILD_ENVIRONMENT: !{{ build_environment }}
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  IN_CI: 1
+  IS_GHA: 1
+  PR_LABELS: ${{ toJson(github.event.pull_request.labels.*.name) }}
+  PR_NUMBER: ${{ github.event.pull_request.number }}
+  PYTORCH_RETRY_TEST_CASES: 1
```
Comment (janeyx99) [inline .github/templates/windows_binary_build_workflow.yml.j2:79]: Do we still want this variable for binary builds? Normally this just allows CI to retry tests and see how flaky they are

Comment (seemethere) [reply .github/templates/windows_binary_build_workflow.yml.j2:79]: We can go ahead and remove in a follow up PR. There might be some cleanup that needs to be done afterwards

**Review by atalman — APPROVED** [review]

**Pull request closed without merging**